WooCommerce/WooCommerceTests/Stripe Integration Tests/StripeCardReaderIntegrationTests.swift

```diff
@@ -0,0 +1,53 @@
+import XCTest
+import Combine
+@testable import WooCommerce
+@testable import Hardware
+
+
+/// Integration tests for the integration with the Stripe Terminal SDK.
+/// We want to reach into the ServiceLocator in all tests
+/// because these are integration test, and we do not want to mock anything,
+/// at this point, other than the actual hardware
+final class StripeCardReaderIntegrationTests: XCTestCase {
+    private var cancellables: Set<AnyCancellable>!
+
+    override func setUp() {
+        super.setUp()
+        cancellables = []
+    }
+
+    /// The observed behaviour is that when discovery starts, the Stripe SDK will
+    /// call into its delegate with an empty array of "discovered readers".
+    /// Later on, it will call into its delegate again everytime it discovers a reader
+    /// passing the full list of readers discovered
+    func test_start_discovery_updates_discovered_readers_at_least_twice() {
+        let discoveredReaders = expectation(description: "Discovered Readers publishes first, an empty array, and then the actual reader(s) discovered")
+
+        let readerService = ServiceLocator.cardReaderService
```

Comment:
I'm guessing this is the actual Stripe card reader service? if so, does it depend on some connection for reader discovery or it's default to run in "test mode" as in the doc?

Reply:
That's correct. For now, it is running, hardcoded, in "test mode". Whenever we switch that flag and move the app into "production mode", we'd have to update these tests to set a card reader service that still attacks the SDK test mode.